### PR TITLE
Use go.mozilla.org package name in CircleCI config

### PR DESCRIPTION
**Pull request by autrilla**
Also updates the Docker image to include the AWS cli and use a newer version of Go

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,25 +1,21 @@
...
+          name: Build containers
+          command: |
+            docker build -t mozilla/sops .
+            docker tag mozilla/sops "mozilla/sops:$CIRCLE_SHA1"
       - run:
           name: Push containers
           command: |
             if [ "${CIRCLE_BRANCH}" == "master" ]; then
                 ./bin/ci/deploy_dockerhub.sh "latest"
                 ./bin/ci/deploy_dockerhub.sh "$CIRCLE_SHA1"
```
Comment (milescrabill) [inline .circleci/config.yml]: This is going to give you a newly tagged image on dockerhub for _every commit_ to master. I suspect you do not want this for something like sops.

File: Dockerfile [hunk 2]
```diff
@@ -1,7 +1,10 @@
...
 WORKDIR /go/src/go.mozilla.org/sops
-RUN go get -d -v
-RUN go test ./...
-RUN go install -v ./...
+
+RUN CGO_ENABLED=0 GOOS=linux go build -a -installsuffix cgo -o /bin/sops ./cmd/sops
+RUN apt-get update
+RUN apt-get install -y vim python-pip emacs
+RUN pip install awscli
+ENV EDITOR vim
```
Comment (milescrabill) [inline Dockerfile]: newline at eof

File: .circleci/config.yml [hunk 3]
```diff
@@ -1,25 +1,21 @@
...
-          name: Build containers                                                                                                                                                                                                                               
-          command: |                                                                                                                                                                                                                                           
-            docker build -t mozilla/sops .                                                                                                                                                                                                          
-            docker tag mozilla/sops "mozilla/sops:$CIRCLE_SHA1"                                                                                                                                                                          
+          name: Build containers
+          command: |
+            docker build -t mozilla/sops .
+            docker tag mozilla/sops "mozilla/sops:$CIRCLE_SHA1"
```
Comment (milescrabill) [inline .circleci/config.yml]: See below. Is this what you want?

File: .circleci/config.yml (hunk 1 above)
Comment (autrilla) [reply .circleci/config.yml]: I think that's fine. We could also tag images with git tags but I don't think pushing on every commit to master would be terrible.

**Pull request merged**